src/utils/warn.ts

```diff
@@ -0,0 +1,9 @@
+export const getLogger = (level: 'log' | 'warn' | 'error') =>
+  process.env.NODE_ENV !== 'production'
+    ? console[level]
+    : () => {
+        // do nothing
+      }
```

Comment:
Unfortunately this makes the types slightly worse than they were before. To get the right types we'll need to do something like this:

```ts
import { isNotProd } from '~/utils/dev'

export const getLogger = <T extends keyof typeof console>(level: T): typeof window.console[T] =>
  isNotProd
    ? console[level]
    : (() => {
        // do nothing
      }) as typeof console[T]

export const warn = getLogger('warn')
export const log = getLogger('log')
```

Reply:
We can probably also rename the module to `console` if it's going to include more than just `warn` :grin:

Reply:
That makes sense, updated in 3ad59c2e.

Reply:
@dhruvkb did you miss https://github.com/WordPress/openverse-frontend/pull/1023#discussion_r816127505 or was it something you think we shouldn't do?

Reply:
This is slightly confusing because I'm still getting the right types. If I import `log` or `warn` from this module they show the same type as `console.log` and `console.warn`.

Reply:
This is the difference I see:

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gSzQBRiWAC54AjJJCEKFAbgCgHl0sBzEDAGSTY5ngBeeAAoA1iACepAOQReM+AB94MgO6wUilTIxQy1GQEohAPgbxL8AIS4CReAH5EqNFRABtCZIC68UiIiJoKm8ADeAL5GTCyY8PJsQvAc3Lz8InIK0QD02VbwAHqOMa5YGjAoSSk8fCAwGeVaOXlWRSWs8HoGCMLVaXUZXYbN+W3tcQBm2AAeIMAA4pw1-EkAPAAq8CDTGCAowGjw3kgT8ADCru6m4lKk60akF+juHut+IRZWtviEwE4uz2oXikfgCQTM4Si8CghyebiBbyYzFK8V4ACYklNZgslv16pk2MYGLl8oVisiOo0McIsXNFqlavjGkSSa1ybEsEMQNT4LScQz0rp9MNiS1LG0gA

I'm not sure exactly why it squashes everything to `(...data: any[]) => void`. I think because it's merging the empty function and the actual console function's type? But it doesn't fix it if you cast the empty function either so I don't think that's why really:

https://www.typescriptlang.org/play?#code/CYUwxgNghgTiAEYD2A7AzgF3gSzQBRiWAC54AjJJCEKFAbgCgHl0sBzEDAGSTY5ngBeeAAoA1iACepAOQReM+AB94MgO6wUilTIxQy1GQEohAPgbxL8AIS4CReAH5EqNFRABtCZIC68UiIiJoKm8ADeAL4mUGjwGJIADiBIAGYu6O4e8Ump8N4+TCyY8PJsQvAc3Lz8InIKRgwA9I1W8AB6joWuWBowKOWVPHwgMLW9Wg3NrR1drHH61AOcQzW6CyDGTS1WM7PFKdgAHiDAAOLL1SPlADwAKvAghxggKMCx3rkAwq7upuJSpFuRlI3wy1A8tz8IQsVls+EIwCc6Tc4Py-lEQTM4Si8Bi8FBKM8kKYzG6JV4ACZygdjmcLsNRnU2JsptNOqS5uMqcIaSdzlUGWNNCztpZdkUsHoDCBufBeXSBaspYZJqL2p0gA

Anyway, it's not a big deal.